src/promptflow/promptflow/exceptions.py

```diff
@@ -209,3 +232,181 @@ class ValidationException(UserErrorException):
     """Exception raised when validation fails."""
 
     pass
+
+
+class ErrorInfo:
+    _error_classes = dict()
+    _exception_codes = dict()
+
+    @classmethod
+    def get_error_info(cls, e: Exception):
+        if not isinstance(e, Exception):
+            return None, None, None, None
+
+        e = cls.select_exception(e)
+        if cls._is_system_error(e):
+            return ErrorCategory.SystemError, cls._error_type(e), cls._error_target(e), cls._error_message(e)
+
+        return ErrorCategory.SDKUserError, cls._error_type(e), cls._error_target(e), cls._error_message(e)
+
+    @classmethod
+    def select_exception(cls, e: Exception):
+        """Select the exception  in e and e.__cause__, and prioritize the Exception defined in the SDK."""
+
+        if e.__cause__ and isinstance(e.__cause__, PromptflowException):
+            return e.__cause__
+
+        if isinstance(e, PromptflowException):
+            return e
+
+        if e.__cause__ and isinstance(e.__cause__, HttpResponseError):
+            return e.__cause__
+
+        return e
+
+    @classmethod
+    def _is_system_error(cls, e: Exception):
+        if isinstance(e, (SystemErrorException, HttpResponseError)):
+            return True
+        if hasattr(e, "status_code") or (hasattr(e, "response") and hasattr(e.response, "status_code")):
+            status_code = str(e.status_code) if hasattr(e, "status_code") else str(e.response.status_code)
+            if not status_code.startswith("40"):
+                return True
+
+        return False
+
+    @classmethod
+    def _error_type(cls, e: Exception):
+        # executor error
+        if isinstance(e, cls.get_error_classes("promptflow.executor._errors")):
```

Comment:
shall we make the info into the error class definition? It looks like many hardcode logic to me here.

Reply:
I have removed this logic, and our "error type" returns "e.__name__".